### ISSUE TITLE
LWP::UserAgent created twice

### ISSUE THREAD

**Issue by XSven** [ByteFoundryAU/perl-WebService-HashiCorp-Vault#3]
The cronstruction of a `WebService::HashiCorp::Vault` object creates an `LWP::UserAgent` object behind the scenes because `WebService::HashiCorp::Vault` has the role of a `WebSerice::Client`. Accesing this user agent with the `ua()` getter and changing something (disabling ssl hostname verification for example) has no effect when accessing a `secret()`. The reason for this is that the `secret()` method constructs a backend (for example `WebService::HashiCorp::Vault::Secret::Generic`) object that has its own `LWP::UserAgent` object. This second user agent object is the one that finally runs the request.  I would suggest to add the following line to the `WebService::HashiCorp::Vault::secret()` method:

```
$args{ua} = $self->ua();  # passes the vault user agent object to the backend
```
This change avoids duplicate user agents and allows a proper user agent tuning before requesting secrets.

**Comment by djzort** [comment]
yes i see what you mean.

See if this commit 8db753335abdb0607ee77d5b5c437e333d7914a7 fixes it for you, and i will happily release to cpan

**Comment by XSven** [comment]
Yes it does. Thanks please feel free to create and upload new CPAN release.

**Comment by djzort** [comment]
ive just released 0.005 to cpan and uploaded here. enjoy
